hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/FixedFileTrailer.java

```diff
@@ -612,6 +612,8 @@ private String getHBase1CompatibleName(final String comparator) {
       comparatorKlass = CellComparatorImpl.class;
     } else if (comparatorClassName.equals(KeyValue.META_COMPARATOR.getLegacyKeyComparatorName())
       || comparatorClassName.equals(KeyValue.META_COMPARATOR.getClass().getName())
+      || (comparatorClassName.equals("org.apache.hadoop.hbase.CellComparator$MetaCellComparator"))      
+      || (comparatorClassName.equals("org.apache.hadoop.hbase.CellComparatorImpl$MetaCellComparator"))      
```

Comment:
Is `CellComparator` not enough? I just checked previous commit where only `CellComparator` was present and not `CellComparatorImpl`

Reply:
Before HBASE-24968, MetaCellComparator.class.getName() was org.apache.hadoop.hbase.CellComparatorImpl$MetaCellComparator